transcom/mymove
# MAIN - B-20602 BE boat shipment customer api

## [B-20602](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20602)

INT PRs: 
https://github.com/transcom/mymove/pull/13370

[#13387](https://github.com/transcom/mymove/pull/13387/files#diff-3bfbe8b71332fe4dc24d752cb37d3463e9d60e2d6250b20e3c5da7f196040995)

https://github.com/transcom/mymove/pull/13398

## Summary

This is to add/modify services and Internal api endpoints related to Boat Shipment on customer's side of the app.

- Added Boat Shipment services(updater, fetcher, creator)
- Updated MTOShipment services to include boat shipment.
- Added boat shipment to MTOShipment internal API(GET, POST, PATCH).
- Feature Flag added to Internal, GHC, and Prime MTOShipment  API Handlers(GET).

### How to test
1. Create a move with any shipment.
2. As a customer ->  click "Go to Move" for the current move.
3. Use the browser dev tools get the info you need for the Swagger or Postman.

Use Swagger or Postman to check Internal APIs:

Import `internal.yaml` file from swagger folder if you have not already.

POST - http://milmovelocal:3000/swagger-ui/internal.html#/mtoShipment/createMTOShipment
Example body to use:
```
{
  "moveTaskOrderID": "a615abd5-c64a-4c0d-b438-2b5e66790130",
  "shipmentType": "BOAT_TOW_AWAY",
  "boatShipment": {
    "type": "TOW_AWAY",
    "year": 1111,
    "model": "TOW_AWAY",
    "make": "TOW_AWAY",
    "lengthInInches": 169,
    "widthInInches": 83,
    "heightInInches": 78,
    "hasTrailer": false
  }
}
```

PATCH - http://milmovelocal:3000/swagger-ui/internal.html#/mtoShipment/updateMTOShipment
Example body to use:
```
{
  "moveTaskOrderID": "2ceb3116-b24b-41df-a023-67ac09ec13de",
  "shipmentType": "BOAT_HAUL_AWAY",
  "boatShipment": {
    "type": "HAUL_AWAY"
  }
}
```

GET - http://milmovelocal:3000/swagger-ui/internal.html#/mtoShipment/listMTOShipments

GET - http://milmovelocal:3000/swagger-ui/internal.html#/moves/getAllMoves


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![image](https://github.com/user-attachments/assets/607aaec9-35e9-4dca-bd95-211c304fa7a5)
![image](https://github.com/user-attachments/assets/2bbf1310-6e17-40b7-9f54-e64bdd1e5fde)
![image](https://github.com/user-attachments/assets/37ad8915-5824-4203-bcfb-0c194ef1f910)
![image](https://github.com/user-attachments/assets/427f0f8a-7621-4bae-b8e0-379a43cbd89f)
